controllers/namespacescope_controller.go

```diff
@@ -1166,6 +1212,86 @@ func (r *NamespaceScopeReconciler) CSVReconcile(req ctrl.Request) (ctrl.Result,
 	return ctrl.Result{RequeueAfter: 180 * time.Second}, nil
 }
 
+func (r *NamespaceScopeReconciler) getWebhooks(csvName string, csvNs string) ([]unstructured.Unstructured, []unstructured.Unstructured, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		klog.Errorf("Failed to get config: %v", err)
+		return nil, nil, err
+	}
+	dynamic := dynamic.NewForConfigOrDie(cfg)
+
+	APIGroup := "admissionregistration.k8s.io"
+	APIVersion := "v1"
+	mKind := "mutatingwebhookconfigurations"
+	vKind := "validatingwebhookconfigurations"
+
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{
+		constant.OlmOwnerLabel:   csvName,
+		constant.OlmKindLabel:    "ClusterServiceVersion",
+		constant.OlmOwnerNsLabel: csvNs,
+	}}
+	// list all the mutatingwebhookconfiguration
+	mutatingwebhookconfigList, err := util.GetResourcesDynamically(ctx, dynamic, APIGroup, APIVersion, mKind, labelSelector)
+	if err != nil {
+		klog.Errorf("error getting resource: %v\n", err)
+		return nil, nil, err
+	}
+	// list all the validatingwebhookconfiguration
+	validatingwebhookconfigList, err := util.GetResourcesDynamically(ctx, dynamic, APIGroup, APIVersion, vKind, labelSelector)
+	if err != nil {
+		klog.Errorf("error getting resource: %v\n", err)
+		return nil, nil, err
+	}
+
+	return mutatingwebhookconfigList, validatingwebhookconfigList, nil
+}
+
+func (r *NamespaceScopeReconciler) patchWebhook(webhook *unstructured.Unstructured, nsList []string) error {
```

Comment:
If we are able to get objectList from function `getWebhooks`, we might not need to deal with `Unstructured` Object, and we could get rid of parsing `interface{}`
- [a list of MutatingWebhookConfiguration object](https://github.com/kubernetes/api/blob/b66611547a244c73e45257add286c7ecc16d5e79/admissionregistration/v1/types.go#L173)
- [a list of ValidatingWebhookConfiguration object](https://github.com/kubernetes/api/blob/v0.27.4/admissionregistration/v1/types.go#L143)